src/ILCompiler.ReadyToRun/src/Compiler/ReadyToRunCompilerContext.cs

```diff
@@ -40,5 +40,24 @@ public override FieldLayoutAlgorithm GetLayoutAlgorithmForType(DefType type)
                 return _r2rFieldLayoutAlgorithm;
             }
         }
+
+        public override void SetSystemModule(ModuleDesc systemModule)
+        {
+            base.SetSystemModule(systemModule);
+
+            // System.Object has an EEType field in the standard AOT version used in this repo.
+            // Make sure that we always use the CoreCLR version which (currently) has no fields.
+            var objectType = systemModule.GetType("System", "Object");
+            Debug.Assert(objectType.GetField("m_pEEType") == null, "Incompatible system library. The CoreCLR System.Private.CoreLib must be used when compiling in ready-to-run mode.");
```

Comment:
I would check this in the `System.Object`-specific field layout algorithm.

The check should ideally be that there's no instance fields. Let's not hardcode `m_pEEType` in more places than needed.

Reply:
Done